semgrep-core/src/engine/Eval_generic.ml

```diff
@@ -191,37 +191,54 @@ let rec eval env code =
 
 and eval_op op values code =
   match (op, values) with
+  | G.And, [ Bool b1; Bool b2 ] -> Bool (b1 && b2)
+  | G.Not, [ Bool b1 ] -> Bool (not b1)
+  | G.Or, [ Bool b1; Bool b2 ] -> Bool (b1 || b2)
   | G.Gt, [ Int i1; Int i2 ] -> Bool (i1 > i2)
-  | G.GtE, [ Int i1; Int i2 ] -> Bool (i1 >= i2)
-  | G.Lt, [ Int i1; Int i2 ] -> Bool (i1 < i2)
-  | G.LtE, [ Int i1; Int i2 ] -> Bool (i1 <= i2)
-  (* todo: we should perform cast and allow to mix Float and Int *)
   | G.Gt, [ Float i1; Float i2 ] -> Bool (i1 > i2)
+  | G.Gt, [ Int i1; Float i2 ] -> Bool (float_of_int i1 > i2)
```

Comment:
We could maybe factorizing code by matching
op, [Int x; Float y] and call recursively eval_op with upcasted Int,
but this is fine too.